webpack/assets/javascripts/react_app/components/AutoCompleteSearch/helpers.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { Menu, MenuItem, Highlighter } from 'react-bootstrap-typeahead';
+
+export const groupBy = (list, keyGetter) => {
+  const map = new Map();
+  list.forEach((item) => {
+    const key = keyGetter(item);
+    const collection = map.get(key);
+    if (!collection) {
+      map.set(key, [item]);
+    } else {
+      collection.push(item);
+    }
+  });
+  return map;
+};
+
+export const renderMenu = (results, menuProps) => {
```

Comment:
This entire filtering would invoked each time the input render.
you already use selectors, is there an option to move some filtering from here ?